packages/txservice/src/monitor/buffer.ts

```diff
@@ -0,0 +1,49 @@
+import { TransactionServiceFailure } from "../error";
+
+import { Transaction } from "./transaction";
+
+/**
+* @classdesc Really basic class: manages a map/buffer like it's an ordered dictionary, essentially.
+*/
+export class TransactionBuffer {
+  // TODO: Maps in TS technically maintain order based on order of insertion - but in order for us to
+  // guarantee that that order will be sequential in terms of nonce, we'd have to enforce that
+  // in set() (in other words, throw if there's an attempt to insert out of order). Is that what
+  // we want to do? If so, orderedKeys() will become obsolete.
+  private buffer: Map<number, Transaction> = new Map();
+
+  public get pending(): Transaction[] {
+    return this.orderedKeys()
+      .map((nonce) => this.get(nonce) ?? null)
+      .filter(tx => tx != null && !tx.didFinish) as Transaction[];
+  }
+
+  public insert(nonce: number, transaction: Transaction, overwrite = false) {
+    // TODO: detect gap in nonce? throw if there's an attempt to insert out of order?
+    // TODO: Trim backend of buffer? Maybe keep a maximum of 1000 transactions?
```

Comment:
Transactions should be deleted from the buffer once they are fully confirmed

Reply:
That does make sense. I'm worried there's an edge case here *somehow* even though I don't see that being possible, so you're right, lets go with that

Reply:
done